### PR TITLE
Tank suicide no longer gibs w/o enough pressure

### PR DESCRIPTION
**Pull request by Cyberboss**
>Make gibs great again

Gibbing might actually mean something if it wasn't available to every player

### CODE REVIEW

**Comment by PKPenguin321** [comment]
incoming very angry players

**Comment by MrStonedOne** [comment]
\>Nerfing a meme during high pop

It's a bold move cotton, lets see how it plays out for him

**Comment by WJohn** [comment]
but why tho

**Comment by RandomMarine** [comment]
I like this if only for that it adds a little layer of realism. But I'd take it a step further and add three stages.
At very low pressures a suicide attempt should fail.
At normal pressures it just succeeds in bursting your lungs.
At high pressures you get the current gibmeme behavior.

**Comment by AnturK** [comment]
Could just remove outright really, but this is fine too.

**Comment by octareenroon91** [comment]
Filling a tank is a pretty low barrier, but go for it

**Comment by TheColdTurtle** [comment]
It would just encourage people to waste air from important tanks, instead of their personal tanks.

**Comment by Cyberboss** [comment]
I could bump it to 1025 which is impossible without an air pump, but I feel
like that's basic Lee removing it.

**Comment by octareenroon91** [comment]
The particularly public tanks I can think of wouldn't miss an emergency-oxygen-tankful of gas.

**Comment by Metallusions** [comment]
Would anyone even notice the difference of going to the air tank in the arrivals closet before suiciding?

**Comment by Cyberboss** [comment]
No tank starts with more than 900kpa

On Apr 26, 2017 9:56 AM, "Metallusions" <notifications@github.com> wrote:

Would anyone even notice the difference of going to the air tank in the
arrivals closet before suiciding?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/tgstation/tgstation/pull/26597#issuecomment-297416388>,
or mute the thread

**Comment by octareenroon91** [comment]
*canister

**Comment by Militaires** [comment]
## but why tho

like, what's the reason behind this

all it feels like is just adding another hoop to jump through before you can suicide

**Comment by Cyberboss** [comment]
Suicide is a means to and end. The way you do it just adds flavor. However,
being easily able to gib yourself is too far, given how rare gibbing is
supposed to be.

**Comment by TalkingCactus** [comment]
Shouldn't this use moles, the actual _amount_ of gas, instead of pressure? 

If my tank was the size of my little finger, but it was at 1000kPa it would still gib me the same as a canister sized tank of 1000kPa with these conditions.

**Comment by Cyberboss** [comment]
From a gameplay perspective that just makes things more complicated
considering you can't actually calculate the moles in a canister without
code diving.

**Comment by Cyberboss** [comment]
In a tank*

**Comment by TalkingCactus** [comment]
`return_volume()`

**Comment by MrStonedOne** [comment]
ya, what he is saying is that a player can not, in game know that amount.

**Comment by RandomMarine** [comment]
On that subject, why don't gas analyzers and such output moles?

**Comment by TalkingCactus** [comment]
Canisters, tanks, and other containers for gasses should also label their volume in their descriptions as well as what RandomMarine said.

Pressure means very little without volume and just because we fail to express volume to the player, when it's used largely in nearly every gas calculation, doesn't mean it should be ignored.

**Comment by lzimann** [comment]
@RandomMarine so you have to calculate it yourself!

**Comment by octareenroon91** [comment]
>If my tank was the size of my little finger, but it was at 1000kPa it would still gib me the same as a canister sized tank of 1000kPa with these conditions.

Sounds like a bullet

File: code/game/objects/items/weapons/tanks/tanks.dm [hunk 1]
```diff
@@ -116,7 +116,7 @@
 	var/mob/living/carbon/human/H = user
 	user.visible_message("<span class='suicide'>[user] is putting [src]'s valve to [user.p_their()] lips! It looks like [user.p_theyre()] trying to commit suicide!</span>")
 	playsound(loc, 'sound/effects/spray.ogg', 10, 1, -3)
-	if (H && !QDELETED(H))
+	if (!QDELETED(H) && air_contents && air_contents.return_pressure() >= 1000)
```
Comment (MrStonedOne) [inline code/game/objects/items/weapons/tanks/tanks.dm]: You need to base this off of the pressure differential. it shouldn't gib you if you do it in 500kpa overpressured room, but it should be easier in a vacuum.

In fact, if you wanted to quiet down some of the criticisms from players you could compromise and make it such that in a vacuum it does gib on stock e-tanks, but not in normal atmosphere.

**Pull request merged**